### DISCUSSION TITLE
dist tag based permissions

### DISCUSSION THREAD

**Discussion by g3hxqe-contrib** [verdaccio/verdaccio#1686] [Ideas]
**Is your feature request related to a problem?**
I would like to allow publishing for authenticated users to publish to "development" dist tag but would like to restrict "latest" tag to be permitted by ci build servers.

**Describe the solution you'd like**
I would like to expose dist tag information to plugins, this way a plugin could implement the authentication logic on weather or not grant permission to the user depending on what tag they are publishing to.

**Describe alternatives you've considered**
I have looked at adding it to the core package authentication configuration. It seems as though it may be confusing to configure. "@myorg/***/@development". It seems like it wouldn't be common so might be better as a plugin, but the data needs to be passed along to the auth plugin api.

**Comment by juanpicado** [comment]
Great idea, just to remark it is only valid for publish and unpublish. But this will be nice for auth plugins to be able create more complex scenarios.
